packages/components/src/RecurringSelect/RecurringSelect.tsx

```diff
@@ -0,0 +1,201 @@
+import { InputNumber } from "../InputNumber";
+import { InputGroup } from "../InputGroup";
+import { Content } from "../Content";
+import { Text } from "../Text";
+
+interface RecurringSelectProps {
+  disabled?: boolean;
+  recurrenceRule: RecurrenceRule;
+  onChange(newPartialState: Partial<SchedulerState>): void;
+}
+
+export function RecurringSelect({
+  recurrenceRule,
+  disabled = false,
+  onChange,
+}: RecurringSelectProps) {
+  function onRecurrenceChange(newRecurrence: { rule: RecurrenceRule }) {
+    onChange({
+      recurrence: {
+        ...newRecurrence,
+      },
+    });
+  }
+  const currentRecurrenceComponent = getCurrentComponent(
+    recurrenceRule,
+    onRecurrenceChange,
+    disabled,
+  );
+  const disabledTextVariation = disabled ? "disabled" : undefined;
+  // we must dynamically populate the select option based on which is selected
+  // because there is no single "month" option, it must always be one of these two
+  const monthlySelectOption =
+    recurrenceRule.type === DurationPeriod.WeekDayOfMonth ? (
+      <Option value={DurationPeriod.WeekDayOfMonth}>Month(s)</Option>
+    ) : (
+      <Option value={DurationPeriod.DayOfMonth}>Month(s)</Option>
+    );
+  const hasExtraFrequencyDescriptor =
+    recurrenceRule.type === DurationPeriod.WeekDayOfMonth ||
+    recurrenceRule.type === DurationPeriod.DayOfMonth ||
+    recurrenceRule.type === DurationPeriod.Week;
+  const isMonthly =
+    recurrenceRule.type === DurationPeriod.WeekDayOfMonth ||
+    recurrenceRule.type === DurationPeriod.DayOfMonth;
+
+  return (
+    <Content>
+      <div className={styles.container}>
+        <Text variation={disabledTextVariation}>Every</Text>
+        <div className={styles.fullWidthWrapper}>
+          <InputGroup flowDirection="horizontal">
+            <InputNumber
+              disabled={disabled}
+              name="schedule-recurrence-interval"
+              value={recurrenceRule.interval}
+              min={1}
+              maxLength={3}
+              onChange={onChangeInterval}
+            />
+
+            <Select
+              disabled={disabled}
+              value={recurrenceRule.type}
+              onChange={onChangeType}
+              name="schedule-recurrence-type"
+            >
+              <Option value={DurationPeriod.Day}>Day(s)</Option>
+              <Option value={DurationPeriod.Week}>Week(s)</Option>
+              {monthlySelectOption}
+              <Option value={DurationPeriod.Year}>Year(s)</Option>
+            </Select>
+          </InputGroup>
+        </div>
+        {hasExtraFrequencyDescriptor && (
+          <Text variation={disabledTextVariation}>on</Text>
+        )}
+      </div>
+      {isMonthly && (
+        <MonthlySelect
+          disabled={disabled}
+          onChange={onChangeType}
+          selectedMonthOption={recurrenceRule.type}
+        />
+      )}
+      <div className={checkBoxStyles.container}>
+        {currentRecurrenceComponent}
```

Comment:
do you feel like this is a nice-to-have, or a must do?

I ask because it's not obvious to me what problem that is solving. eg. is it to follow best practices, to make the component smaller, more readable etc.

Reply:
@ZakaryH you pretty much answered your question there haha. all of the above :P